### PR TITLE
Fix firewall rule management.

### PR DESCRIPTION
**Pull request by DFINITYManu**
* Fix typo in proposal to remove firewall rules.
* Fix signature of remove command being called.
* Batch multiple operations so we can at least remove / modify / add multiple rules in one proposal.
* Add standard title, summary and motivation parameters to firewall management.

Before we do the batching, we sort the rules by modification slot
then we reverse the order of the rules list, and then we do the 
batching.  This ensures that the modifications happen in an order
that will not conflict between each other (what does it mean "add 
rule in position 3" if rule in position 2 was removed prior to 
that?).

### CODE REVIEW

**Review by NikolaMilosa — APPROVED** [review]
Ah sorry. Great catch!

**Comment by DFINITYManu** [comment]
@sasa-tomic @NikolaMilosa PTAL.

File: rs/cli/src/ic_admin.rs
Comment (sasa-tomic) [inline rs/cli/src/ic_admin.rs:810]: remove?

**Review by sasa-tomic — APPROVED** [review]
LGTM, but the complexity grew substantially compared to the original PR. I don't see an obvious issue, but this is a critical piece of configuration so IMHO it would be good to at least run this on a testnet to verify that the code would work as expected.

File: rs/cli/src/ic_admin.rs
Comment (DFINITYManu) [reply rs/cli/src/ic_admin.rs:810]: I'll add that this change does not impact any critical piece of configuration.  This change merely modifies how we call `ic-admin` and what data we pass onto it.  I want to be very clear about the scope of this change:

There are two ways in which someone could maliciously or accidentally operate this tool in which it could hypothetically break the IC.  The operator -- or the tool -- could enter an invalid payload (junk), or he could enter a vexatious yet valid payload (e.g. "block all connections from all IPs")

W.r.t invalid payloads:

* The correct place to establish safeguards as to what critical data and code is valid is on the server side (the replica), and the safeguards are already established there plus they are well-tested.
* The second place where safeguards should be in place is `ic-admin` itself (since `dre` merely invokes it here).  I am happy to report that I have personally tested `ic-admin` refuses to load invalid payloads, so the safeguards are already in place.

W.r.t vexatious payloads:

Nothing we can do from the perspective of maintainers of `dre`.  If someone sends a proposal that wrecks the IC, and the proposal gets voted in, then that's not a responsibility of the authors of the `dre` tool.

We could hypothetically add some heuristic that prevents extreme breakage (detect overly zealous block rules, like `rm -rf /` does today).  That, however, would not make this specific change any safer -- it is an engineering decision entirely orthogonal to this change.

**Comment by DFINITYManu** [comment]
After merge this will be tested in staging.

**Pull request merged**
